pyproject.toml

```diff
@@ -3,7 +3,7 @@
 # See https://github.com/scipy/scipy/pull/12940 for the AIX issue.
 requires = [
     "meson-python>=0.13.1",
-    "meson>=1.2.1,<2",
+    "meson>=1.2.1",
```

Comment:
I removed the <2 as I didn't want to repeat it everywhere in our CI configurations, and we don't create upper bounds like this on any other dependency. There also is not a meson 2.0 under development